### PR TITLE
Fix route to RN conversation details in Teacher

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-14257
affects: Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/lh261af) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Flh261af)
teacher: [Link](https://rebrand.ly/z1tfnbz) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fz1tfnbz)
parent: [Link](https://rebrand.ly/s1kgxo0) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fs1kgxo0)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher

[MBL-14257](https://instructure.atlassian.net/browse/MBL-14257)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.65% | 94.23% | -0.59%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: CanvasCore/CanvasCore/Helm/Helm.swift [hunk 1]
```diff
@@ -262,6 +262,10 @@ open class HelmManager: NSObject {
         if let navigationController = topViewController.navigationController {
             pushOntoNav(navigationController)
             callback?()
+        }
+        else if let split = topViewController as? UISplitViewController, let navigationController = split.masterNavigationController {
```
Comment (bluwave) [inline CanvasCore/CanvasCore/Helm/Helm.swift:266]: This is needed on iPhone or it hits the assertion lower down
```
assertionFailure("\(#function) invalid controller: \(topViewController)")
```

**Comment by bluwave** [comment]
Sorry to force push, had to get rid of the other Helm changes.

**Pull request merged**